cdc/changefeed.go

```diff
@@ -847,6 +895,103 @@ func (c *changeFeed) pullDDLJob() error {
 	return nil
 }
 
+// startSyncPeriod start a timer for every changefeed to create sync point by time
+func (c *changeFeed) startSyncPeriod(ctx context.Context, interval time.Duration) {
+	log.Debug("sync ticker start", zap.Duration("sync-interval", interval))
+	go func(ctx context.Context) {
+		ticker := time.NewTicker(interval)
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case <-ticker.C:
+				c.updateResolvedTs = false
+			}
+		}
+	}(ctx)
+}
+
+//createSynctable create a sync table to record the
+func (c *changeFeed) createSynctable(ctx context.Context) error {
+	database := "TiCDC"
+	tx, err := c.syncDB.BeginTx(ctx, nil)
+	if err != nil {
+		log.Info("create sync table: begin Tx fail")
+		return err
+	}
+	_, err = tx.Exec("CREATE DATABASE IF NOT EXISTS " + database)
+	//_,err := c.syncDB.Exec("CREATE DATABASE IF NOT EXISTS " + database)
+	if err != nil {
+		err2 := tx.Rollback()
+		if err2 != nil {
+			log.Error(err2.Error())
+		}
+		return err
+	}
+	_, err = tx.Exec("USE " + database)
+	if err != nil {
+		err2 := tx.Rollback()
+		if err2 != nil {
+			log.Error(err2.Error())
+		}
+		return err
+	}
+	_, err = tx.Exec("CREATE TABLE  IF NOT EXISTS syncpoint (cf varchar(255),primary_ts varchar(18),secondary_ts varchar(18),PRIMARY KEY ( `primary_ts` ) )")
```

Comment:
`` PRIMARY KEY ( `primary_ts` ) ``
multiple changefeeds could have the same `primary_ts`, what about using `` PRIMARY KEY (`cf`, `primary_ts`) ``